### PR TITLE
Added support for auto-tracking personalization in Adobe Journey Optimizer.

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

This adds a configuration in the extension configuration, and an option in the apply propositions command.

## Screenshots (if appropriate):
Screenshot 2024-05-10 at 4 09 00 PM
Screenshot 2024-05-10 at 4 09 37 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.
- [ ] I've updated the schema in extension.json or no changes are necessary.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/view/actions/applyPropositions.jsx [hunk 1]
```diff
@@ -50,20 +88,59 @@ const applyPropositionsForm = form({}, [
...
-        name: "selector",
-        label: "Selector",
-        description: "Enter your selector",
-        validationSchemaBase: string().required("Please provide a selector.")
+      radioGroup({
+        name: "elementType",
+        label: "Target element",
+        items: [
+          { value: "selector", label: "Selector" },
+          { value: "element", label: "Element" }
```
Comment (jasonwaters) [inline src/view/actions/applyPropositions.jsx:96]: The `trackProposition` command did support an element or selector to be passed in, but `applyPropositions` does not support this.  It relies on selectors only.  So element should be removed here.

File: src/view/configuration/personalizationSection.jsx [hunk 2]
```diff
@@ -22,24 +23,34 @@ import copyPropertiesWithDefaultFallback from "./utils/copyPropertiesWithDefault
...
     prehidingStyle: "",
     targetMigrationEnabled: false,
-    personalizationStorageEnabled: false
+    personalizationStorageEnabled: false,
+    autoTrackPropositionInteractions: "always"
   }),
   getInitialInstanceValues: ({ instanceSettings }) => {
     const instanceValues = {};
 
+    if (instanceSettings.autoTrackPropositionInteractions?.AJO) {
```
Comment (jasonwaters) [inline src/view/configuration/personalizationSection.jsx:38]: How can a customer enable `autoTrackPropositionInteractions` for Target if they want to in 6 months?

**Review by jasonwaters — COMMENTED** [review]
Looks good @jonsnyder , left a few comments for you.

BTW, I have update my pr to remove `trackProposition` and and enhanced `applyPropositions` to match what you've done here.

File: src/view/configuration/personalizationSection.jsx (hunk 2 above)
Comment (jonsnyder) [reply src/view/configuration/personalizationSection.jsx:38]: They wait for a Web SDK update.

**Comment by jonsnyder** [comment]
Closed in favor of #409

**Pull request closed without merging**